packages/protocol-contracts/test/ZetaTokenConsumer.spec.ts

```diff
@@ -0,0 +1,256 @@
+    return block.timestamp;
+  };
+
+  const swapToken = async (signer: SignerWithAddress, tokenAddress: string, expectedAmount: BigNumber) => {
+    const uniswapRouter = UniswapV2Router02__factory.connect(uniswapV2RouterAddr, signer);
+
+    const WETH = await uniswapRouter.WETH();
+    const path = [WETH, tokenAddress];
+    const tx = await uniswapRouter
+      .connect(signer)
+      .swapETHForExactTokens(expectedAmount, path, signer.address, (await getNow()) + 360, { value: parseEther("10") });
+
+    await tx.wait();
+  };
+
+  /**
+   * @custom:todo (andy): WIP, not in use yet
+   */
+  const createPoolV3 = async (signer: SignerWithAddress, tokenAddress: string) => {
+    await swapToken(signer, DAI, parseUnits("10000", 18));
+
+    const token = IERC20__factory.connect(USDC, signer);
+    const tx1 = await token.approve(UNI_NFT_MANAGER_V3, MaxUint256);
+    await tx1.wait();
+
+    const token2 = IERC20__factory.connect(DAI, signer);
+    const tx2 = await token2.approve(UNI_NFT_MANAGER_V3, MaxUint256);
+    await tx2.wait();
+
+    const uniswapRouter = INonfungiblePositionManager__factory.connect(UNI_NFT_MANAGER_V3, signer);
+
+    const uniswapNFTManager = IPoolInitializer__factory.connect(UNI_NFT_MANAGER_V3, signer);
+    const tx3 = await uniswapNFTManager.createAndInitializePoolIfNecessary(
+      USDC,
+      DAI,
+      3000,
+      "80000000000000000000000000000"
+    );
+    await tx3.wait();
+
+    const params: INonfungiblePositionManager.MintParamsStruct = {
+      token0: USDC,
+      token1: DAI,
+      fee: 3000,
+      tickLower: 193,
+      tickUpper: 194,
+      amount0Desired: parseEther("10"),
+      amount1Desired: parseEther("10"),
+      amount0Min: 0,
+      amount1Min: 0,
+      recipient: signer.address,
+      deadline: (await getNow()) + 360,
+    };
+
+    const tx4 = await uniswapRouter.mint(params);
+    await tx4.wait();
+  };
+
+  beforeEach(async () => {
+    accounts = await ethers.getSigners();
+    [tssUpdater, tssSigner, randomSigner] = accounts;
+
+    zetaTokenNonEth = await deployZetaNonEth({
+      args: [tssSigner.address, tssUpdater.address],
+    });
+
+    uniswapV2RouterAddr = getAddress("uniswapV2Router02", {
+      customNetworkName: "eth-mainnet",
+      customZetaNetwork: "mainnet",
+    });
+
+    // dev: for testing purposes we use an existing uni v3 pool
+    await swapToken(tssUpdater, DAI, parseEther("10000"));
+    await swapToken(randomSigner, DAI, parseEther("10000"));
+    await swapToken(randomSigner, DAI, parseEther("10000"));
+    await swapToken(randomSigner, DAI, parseEther("10000"));
+    await swapToken(randomSigner, DAI, parseEther("10000"));
+    zetaTokenNonEthAddress = DAI;
+    zetaTokenNonEth = IERC20__factory.connect(zetaTokenNonEthAddress, tssSigner);
+
+    zetaTokenConsumerUniV2 = await getZetaTokenConsumerUniV2Strategy({
+      deployParams: [zetaTokenNonEthAddress, uniswapV2RouterAddr],
+    });
+
+    uniswapV3RouterAddr = UNI_ROUTER_V3;
```

Comment:
We can add this address to our addresses e.g.: `npx ts-node src/add-new-address uniswapV3Pool` (from the addresses folder)